tests/testthat/test-map-depth.R

```diff
@@ -8,11 +8,21 @@ test_that("map_depth modifies values at specified depth", {
   expect_equal(map_depth(x1, 2, length), list(list(2)))
   expect_equal(map_depth(x1, 3, length), list(list(list(3, 3))))
   expect_equal(map_depth(x1, -1, length), list(list(list(3, 3))))
-  expect_equal(map_depth(x1, 4, length), list(list(list(list(1, 1, 1), list(1, 1, 1)))))
   expect_snapshot(map_depth(x1, 6, length), error = TRUE)
   expect_snapshot(map_depth(x1, -5, length), error = TRUE)
 })
 
+test_that("default doesn't recurse into data frames, but can customise", {
+  x <- list(data.frame(x = 1), data.frame(y = 2))
+  expect_error(map_depth(x, 2, class), "not deep enough")
+
+  x <- list(data.frame(x = 1), data.frame(y = 1))
+  expect_equal(
+    map_depth(x, 2, class, .is_leaf = Negate(is.list)),
```

Comment:
The use of `Negate` makes me wonder if it wouldn't be better to take `is_node` instead of `is_leaf`?

Reply:
Yeah, I wondered about this too. I might do it in a separate PR.